teamvault/apps/secrets/views.py

```diff
@@ -168,9 +192,31 @@ def form_valid(self, form):
             if attr in form.cleaned_data:
                 setattr(secret, attr, form.cleaned_data[attr])
         secret.save()
-
-        if secret.content_type == Secret.CONTENT_PASSWORD and form.cleaned_data['password']:
-            plaintext_data = form.cleaned_data['password']
+        plaintext_key = None
+        if secret.content_type == Secret.CONTENT_PASSWORD:
+            if form.cleaned_data['password']:
+                plaintext_data = form.cleaned_data['password']
+            else:
+                plaintext_data = None
+            if form.cleaned_data['otp_key_data']:
+                plaintext_key_data = form.cleaned_data['otp_key_data']
+                key_index = plaintext_key_data.index("secret") + 7
+                plaintext_key = plaintext_key_data[
+                                key_index:
+                                plaintext_key_data[key_index:].index("&") + key_index
+                                ]
+                digits = 8 if "digits=8" in plaintext_key_data else 6
```

Comment:
Hier wird noch mal geparst und das selbe JSON zusammengebaut wie oben -> in eine Funktion auslagern